xmpp_information_query.go

```diff
@@ -15,12 +15,18 @@ func (c *Client) Discovery() (string, error) {
 	return c.RawInformationQuery(c.jid, c.domain, reqID, IQTypeGet, XMPPNS_DISCO_ITEMS, "")
 }
 
-// Discover information about a node
+// Discover information about a node. Empty node queries info about server itself.
 func (c *Client) DiscoverNodeInfo(node string) (string, error) {
 	query := fmt.Sprintf("<query xmlns='%s' node='%s'/>", XMPPNS_DISCO_INFO, node)
 	return c.RawInformation(c.jid, c.domain, "info3", IQTypeGet, query)
 }
 
+// Discover information about given item from given jid.
+func (c *Client) DiscoverInfo(from string, to string) (string, error) {
```

Comment:
What's the reason to be able to define the `from`?

Reply:
To be able to do [this](https://github.com/aleesa-bot/aleesa-jabber-go/blob/189cb99ed9ff38237e6a1cdfae2cec5fd76282da/lib.go#L148). Query info about whether client is able to join muc or not, just before blindly try to join.

Reply:
But why not use c.Jid, e.g. as done [here](https://github.com/xmppo/go-xmpp/pull/151/files#diff-613ad8687c9063077397c389e6a49e26e2ecd64207d7ec0472554420d2f82442R21) instead of making it configurable?

Is there any reason that you might have an connection of jid1 that would request this info claiming to be jid2? Actually I'd assume the server would not allow you to impersonate another JID, therefore I don't see why this should be made configurable.

Reply:
It was about year ago, so i don't quite remember why, to be honest. But this mechanics works in conjunction with jabber.ru server and I have to use it _this_ way, otherwise I did not suggest it as PR. If you see it questionable you can always close this PR.

Reply:
But in the example you linked me you also set from to your own JID (I'm pretty sure faking a JID wouldn't work anyway). So if you don't make it configurable but use c.jid, like DiscoverNodeInfo() does, your code should also work.

Reply:
As I am still not convinced it makes sense to have the from configurable I merged that PR and just changed it to always use the own JID.